middle.mkd

```diff
@@ -274,8 +274,9 @@ A GeoJSON object with the type "Feature" is a feature object.
 A GeoJSON object with the type "FeatureCollection" is a feature
 collection object. A feature collection object MUST have a member with
 the name "features". The value of "features" is an array. Each element
-of the array is a feature object as defined above. It is possible for
-this array to be empty.
+of the array MUST be a feature object as defined above. It is possible
+for this array to be empty. The values in the array are ordered, meaning
+that GeoJSON implementations MUST maintain order when processing them.
```

Comment:
On 2016-02-17 23:42, Allan Doyle wrote:
> What if I want to build a GeoJSON implementation that sorts features in
> a feature collection. If someone comes along in a few years, doesn't
> understand the history, and makes a conformance test for GeoJSON
> processors, then my sorter fails.

yes, and it should (if you talk about sorting a collection and then 
claiming that you haven't changed it). if sequence matters, then sorting 
changes what the GeoJSON represents, and thus your sorter isn't just 
doing something lexical, it's changing the meaning of the GeoJSON.

if on the other hand we had decided that a FeatureCollection is 
unordered, then implementing your sorter would be a safe thing to do, as 
it is not changing the meaning of the GeoJSON.

i think that's an excellent example. these are the typical questions 
developers have, and the spec should give them answers.

if we don't say one way or another, devs are left guessing, and some 
might go with the "it's called a collection, so it's unordered" 
reasoning, and others with the "but it's a JSON array, so it's ordered" 
point of view. and both would be kind of right but nobody really knows.

from a long painful history of other formats, we know that this is not 
good. it means that different implementations implement different 
models, and that harms interop.